### PR TITLE
Peter dev

### PR DESCRIPTION
**Pull request by kretep**
Endpoint refactoring, introduction of master endpoint, automatic connecting to master endpoint, introduction of configuration in json file and environment parameter to switch between configurations.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,6 +36,15 @@ $ gulp serve
...
 More information on the gulp tasks in [this README.md](app/templates/gulp/README.md).
 
+### Configuration for development, testing and production
+
+There are different configurations available for development, testing and production:
+'dev', 'test' and 'prod'. Specify your environment by passing a parameter to gulp
+('dev' is the default):
+* 'gulp --env dev serve'
+* 'gulp --env test test'
+* 'gulp --env prod build'
```
Comment (rnugraha) [inline README.md]: Just to my understanding, so now we need to specify the env in order to execute? If it's yes, we might need to adjust Bamboo.

File: src/app/connections/connections.controller.js [hunk 2]
```diff
@@ -20,67 +20,58 @@ angular.module('transmartBaseUi')
       // Predefined endpoints
       $scope.connections = [
         {title: 'transmart-gb', url: 'http://transmart-gb.thehyve.net/transmart', isOAuth:  true},
-        //{title: 'localhost', url: 'http://localhost:8080/transmart', isOAuth:  true}
+        {title: 'localhost', url: 'http://localhost:8080/transmart', isOAuth:  true}
```
Comment (rnugraha) [inline src/app/connections/connections.controller.js]: can we remove this so that is not displayed in test env

File: README.md (hunk 1 above)
Comment (kretep) [inline README.md]: It's not compulsory. 'Dev' is the default.

File: src/app/connections/connections.controller.js (hunk 2 above)
Comment (kretep) [inline src/app/connections/connections.controller.js]: Ok, done.

**Pull request merged**